kafka-admin/src/main/java/org/bf2/admin/kafka/admin/handlers/RequestRewriter.java

```diff
@@ -0,0 +1,38 @@
+package org.bf2.admin.kafka.admin.handlers;
+
+import io.quarkus.vertx.web.RouteFilter;
+import io.vertx.ext.web.RoutingContext;
+import org.bf2.admin.kafka.admin.HttpMetrics;
+import org.jboss.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class RequestRewriter {
+
+    private static final Logger LOG = Logger.getLogger(RequestRewriter.class);
+    private static final String REST = "/rest";
+    private static final String OPENAPI = "/openapi";
+
+    @Inject
+    HttpMetrics httpMetrics;
+
+    @RouteFilter(400)
+    void filterRequest(RoutingContext context) {
+        String requestUri = context.request().uri();
+
+        if (requestUri.startsWith(REST)) {
+            httpMetrics.getDeprecatedRequestCounter(requestUri).increment();
+
+            String remainingPath = requestUri.substring(REST.length());
+            String target = remainingPath.startsWith(OPENAPI) ? remainingPath : "/api/v1" + remainingPath;
```

Comment:
why not check against the string `/rest/` to avoid any potential conflicts with naming? sure this defined API so the chance of getting there is zero. 

Remember from the previous life, rewriting the URL was sometimes not enough as context processing would have already happened, where I had either copy the context-aware properties then forward or do 302. maybe this engine this works fine

Reply:
@rareddy do you mean to avoid mapping something like `/restsomething` to `/api/v1something` ? I think since we know the set of valid paths, either way will safely result in a 404 to the client.

Reply:
right, the risk is zero